From 59f88dbf7f91bcb0e69792b685a7a9ff34e09e10 Mon Sep 17 00:00:00 2001
From: QbProg <tholag@gmail.com>
Date: Sat, 19 May 2012 16:11:41 +0200
Subject: [PATCH] [warning-fix] Duplicate checks removed This includes && with
 always true vars, duplicate checks in a series cascaded
 "if", duplicate function calls in a condition

---
 src/STEPCAFControl/STEPCAFControl_Reader.cxx |    1 -
 src/Standard/Standard_GUID.cxx               |    6 +++---
 src/TObj/TObj_Model.cxx                      |    2 +-
 src/ViewerTest/ViewerTest.cxx                |    1 -
 src/ViewerTest/ViewerTest_ViewerCommands.cxx |    6 +-----
 5 files changed, 5 insertions(+), 11 deletions(-)

Index: oce/src/STEPCAFControl/STEPCAFControl_Reader.cxx
===================================================================
--- oce.orig/src/STEPCAFControl/STEPCAFControl_Reader.cxx
+++ oce/src/STEPCAFControl/STEPCAFControl_Reader.cxx
@@ -938,7 +938,6 @@ static TDF_Label GetLabelFromPD (const H
   S = TransferBRep::ShapeResult ( TP, binder );
   if ( S.IsNull() ) return L;
 
-  if ( S.IsNull() ) return L;
   if ( ShapeLabelMap.IsBound ( S ) )
     L = ShapeLabelMap.Find ( S );
   if ( L.IsNull() )
Index: oce/src/Standard/Standard_GUID.cxx
===================================================================
--- oce.orig/src/Standard/Standard_GUID.cxx
+++ oce/src/Standard/Standard_GUID.cxx
@@ -114,7 +114,7 @@ Standard_Boolean Standard_GUID::CheckGUI
   if (strlen(aGuid) == Standard_GUID_SIZE) {
     Standard_Integer i;
 
-    for (i = 0; i < 8 && result; i++) {
+    for (i = 0; i < 8 ; i++) {
       if (!IsXDigit(aGuid[i])) {
 	return Standard_False;
       }
@@ -122,7 +122,7 @@ Standard_Boolean Standard_GUID::CheckGUI
 
     if (aGuid[8] != '-') return Standard_False;
 
-    for (i = 9; i < 13 && result; i++) {
+    for (i = 9; i < 13; i++) {
       if (!IsXDigit(aGuid[i])) {
 	return Standard_False;
       }
@@ -130,7 +130,7 @@ Standard_Boolean Standard_GUID::CheckGUI
 
     if (aGuid[13] != '-') return Standard_False;
     
-    for (i = 14; i < 18 && result; i++) {
+    for (i = 14; i < 18; i++) {
       if (!IsXDigit(aGuid[i])) {
 	return Standard_False;
       }
